internal/store/nats.go

```diff
@@ -174,9 +175,58 @@ func (n *natsStore) GetServer(ctx context.Context, serverID uuid.UUID) (*model.S
 	return &model.Server{ID: obj.UUID, FacilityCode: obj.FacilityCode}, nil
 }
 
-// List is deprecated and is not implemented here
-func (n *natsStore) List(_ context.Context, _ uuid.UUID, _ rctypes.State) ([]*rctypes.Condition, error) {
-	return nil, errors.New("unimplemented")
+// List retrieves any active conditions
+func (n *natsStore) List(ctx context.Context, srvID uuid.UUID, incState rctypes.State) ([]*rctypes.Condition, error) {
+	_, span := otel.Tracer(pkgName).Start(ctx, "NatsStore.List")
+	defer span.End()
+	le := n.log.WithFields(logrus.Fields{
+		"serverID": srvID.String(),
+	})
+
+	// ugh, so much copy-pasta
+	kve, err := n.bucket.Get(srvID.String())
+	switch {
+	case err == nil:
+	case errors.Is(nats.ErrKeyNotFound, err):
+		le.Debug("no active condition for device")
+		return nil, ErrConditionNotFound
+	default:
+		natsError("get")
+		le.WithError(err).Warn("looking up active condition")
+		return nil, errors.Wrap(ErrRepository, err.Error())
+	}
+
+	var cr conditionRecord
+	if err := cr.FromJSON(kve.Value()); err != nil {
+		le.WithError(err).Warn("bad condition data")
+		return nil, errors.Wrap(ErrRepository, err.Error())
+	}
+
+	var found bool
+	var outgoing rctypes.Condition
+	for _, c := range cr.Conditions {
+		//nolint:gocritic // inverting this test is a moronic suggestion and you should feel bad about it
+		if !rctypes.StateIsComplete(c.State) {
+			// The first incomplete condition is good enough
+			found = true
```

Comment:
instead of assigning and breaking out here, could we `return  []*rctypes.Condition{&outgoing}, nil`

Reply:
The current handler inspects the condition, because of the way Serverservice handles conditions (that is, as versioned attributes). The handler is too smart here, but we can’t address that until we can swap NATS in for the other repository.